### ISSUE TITLE
Refactor Blade component children

### ISSUE THREAD

**Issue by zepfietje** [laravel-filament/filament#13730]
E.g. remove icon prop from button and simply pass an icon component.
Same for dropdown list item badge and probably others.

**Comment by danharrin** [comment]
@zepfietje Can you explain the benefits here please? I think I agree about badges, but I think the icons should remain as they are (I encourage you to take a look at how icons are output in v4 too)

**Comment by danharrin** [comment]
Badges refactored in #15506, I can't see other places where this would be useful but open to suggestions.

**Comment by zepfietje** [comment]
The current system supports only a single icon in a specified position.
Passing an icon element as child of the button component would allow for much more flexible use in Blade and just feels more like HTML.

**Comment by danharrin** [comment]
Yeah, there are lots of considerations that would make this tricky though. For example on a button, the icon needs to be replaced with a loading indicator when loading
